### PR TITLE
[check-mysql] add readonly subcommand

### PR DESCRIPTION
**Pull request by ichirin2501**
check-mysqlでreadonlyの監視機能を追加してみました

### CODE REVIEW

File: check-mysql/lib/readonly.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"os"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/mackerelio/checkers"
+)
+
+type readOnlyOpts struct {
+	mysqlSetting
```
Comment (astj) [inline check-mysql/lib/readonly.go]: It's better to accept options to specify which status to expect, IS readonly or IS NOT readonly.

Comment (ichirin2501) [reply check-mysql/lib/readonly.go]: I see now.
I added a flag parameters and committed it. https://github.com/mackerelio/go-check-plugins/pull/205/commits/b25ceaedecc4a5f01102186752bc406c89a3ea5c

File: check-mysql/lib/readonly.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	NoReadOnly bool `short:"n" long:"no-readonly" description:"expect that it is not read_only with the MySQL parameter"`
+}
+
+func isReadOnly(v string) bool {
```
Comment (itchyny) [inline check-mysql/lib/readonly.go]: This function looks weird to me... maybe inlining is better for this...

**Comment by itchyny** [comment]
How about this? `check-mysql readonly on`, `check-mysql readonly off`

File: check-mysql/lib/readonly.go (hunk 2 above)
Comment (ichirin2501) [reply check-mysql/lib/readonly.go]: ON / OFF of character string is dependent on MySQL side. On the program side I had the reason that I wanted to convert it once and treat it as True / False.

There is no strong reason so I will fix it. (inlining)

File: check-mysql/lib/readonly.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	rows, res, err := db.Query("SHOW GLOBAL VARIABLES LIKE 'read_only'")
+	if err != nil {
+		return checkers.Unknown("couldn't execute query")
+	}
+
+	idxReadOnly := res.Map("Value")
+	readOnlyStatus := rows[0].Str(idxReadOnly)
+
+	if isReadOnly(readOnlyStatus) == opts.NoReadOnly {
+		return checkers.Critical("the expected value of read_only is different")
```
[the remark this answers is not in the document]
Comment (ichirin2501) [reply check-mysql/lib/readonly.go]: ok !!

**Comment by ichirin2501** [comment]
> How about this? `check-mysql readonly on`, `check-mysql readonly off`

Wow, that is simple and good. I would like to do so with arguments instead of options.

**Comment by ichirin2501** [comment]
fix commit! https://github.com/mackerelio/go-check-plugins/pull/205/commits/e31287fd4c2f9bf7fe4ca3133b0a6f6c9f924d41

**Comment by astj** [comment]
Thank you!

**Pull request merged**